ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/util/EpochProcessorUtil.java

```diff
@@ -600,43 +565,37 @@ public static Boolean shouldUpdateValidatorRegistry(BeaconState state)
    */
   public static void update_validator_registry(BeaconState state) throws EpochProcessingException {
     try {
-      UnsignedLong currentEpoch = BeaconStateUtil.get_current_epoch(state);
+      long currentEpoch = BeaconStateUtil.get_current_epoch(state);
       List<Integer> active_validators =
           ValidatorsUtil.get_active_validator_indices(state.getValidator_registry(), currentEpoch);
-      UnsignedLong total_balance = BeaconStateUtil.get_total_balance(state, active_validators);
+      long total_balance = BeaconStateUtil.get_total_balance(state, active_validators);
 
-      UnsignedLong max_balance_churn =
-          BeaconStateUtil.max(
-              UnsignedLong.valueOf(Constants.MAX_DEPOSIT_AMOUNT),
-              total_balance.dividedBy(
-                  UnsignedLong.valueOf((2 * Constants.MAX_BALANCE_CHURN_QUOTIENT))));
+      long max_balance_churn =
+          Math.max(
+              Constants.MAX_DEPOSIT_AMOUNT,
+              total_balance / 2 * Constants.MAX_BALANCE_CHURN_QUOTIENT);
```

Comment:
The order of operations here is changed as compared to the above. It's worth double checking the spec, but what was written above was `total_balance / (2 * Constants.MAX_BALANCE_CHURN_QUOTIENT));`